core/bin/zksync_server/src/main.rs

```diff
@@ -37,6 +37,10 @@ struct Cli {
     /// Generate genesis block for the first contract deployment using temporary DB.
     #[arg(long)]
     genesis: bool,
+    /// Wait for the `setChainId` event during genesis.
+    /// If `--genesis` is not set, this flag is ignored.
+    #[arg(long)]
+    set_chain_id: bool,
```

Comment:
why is it configueable/optional? Is it just for the time being?

Reply:
Firstly, there's no need for it while we don't have a shared bridge. Secondly, with the shared bridge setup, we already run `zk server --genesis` twice (first time only to generate commitments, before contract deployment, IIRC @kelemeno?). During the second time, we will run it with `--set-chain-id`.